### PR TITLE
PCHR-808: Fixed pay field not shown on 'job contract' tab

### PR DESCRIPTION
**Pull request by Agares4**
The "Pay" field on contact summary page was not properly loaded before, and did not appear on the contract page, this PR fixes this issue.

### CODE REVIEW

File: hrjobcontract/js/src/job-contract/controllers/contract.js [hunk 1]
```diff
@@ -107,19 +107,28 @@ define([
...
-                    results.health = health;
+                });
 
-                    return results;
+                var promisePay = ContractPayService.getOne({
+                    jobcontract_revision_id: results.details.jobcontract_revision_id
                 });
+
+                return $q.all({
+                        pay: promisePay,
```
Comment (AkA84) [inline hrjobcontract/js/src/job-contract/controllers/contract.js]: The indentation looks strange, are you using 4 spaces?

Comment (Agares4) [inline hrjobcontract/js/src/job-contract/controllers/contract.js]: yes, should I use two?

File: hrjobcontract/js/src/job-contract/controllers/contract.js [hunk 2]
```diff
@@ -107,19 +107,28 @@ define([
...
+                        health: promiseHealth
+                    })
+                    .then(function(additionalResults) {
```
Comment (AkA84) [inline hrjobcontract/js/src/job-contract/controllers/contract.js]:
```js
.then(function (additionalResults) {
```

please make sure to follow this code style

**Comment by AkA84** [comment]
> The "Pay" field on contact summary page was not properly loaded before

Why?

> this PR fixes this issue.

How?

**Comment by Agares4** [comment]
It was not loaded because wrong argument was sent to it (jobcontract_id instead of jobcontract_revision_id).
The fix uses jobcontract_revision_id.

File: hrjobcontract/js/src/job-contract/controllers/contract.js [hunk 3]
```diff
@@ -105,18 +105,27 @@ define([
...
+
+                return $q.all({
+                    pay: promisePay,
+                    health: promiseHealth
                 })
-                .then(function(health) {
-                    results.health = health;
+                .then(function (additionalResults) {
+                    results.pay = additionalResults.pay;
+                    results.health = additionalResults.health;
```
Comment (AkA84) [inline hrjobcontract/js/src/job-contract/controllers/contract.js]: Would this work (slightly less verbose)?
```js
define([
    'common/angular',
    'job-contract/controllers/controllers',
    // ...
], function (angular, controllers) {
    // ...

    $q.all({
        details: ContractDetailsService.getOne({ jobcontract_id: contractId }),
        hour: ContractHourService.getOne({ jobcontract_id: contractId }),
        leave: ContractLeaveService.get({ jobcontract_id: contractId }),
        pension: ContractPensionService.getOne({ jobcontract_id: contractId })
    })
    .then(function (results) {
        return $q.all({
            pay: ContractPayService.getOne({
                jobcontract_revision_id: results.details.jobcontract_revision_id
            }),
            health: ContractHealthService.getOne({
                jobcontract_revision_id: results.details.jobcontract_revision_id
            })
        })
        .then(function (additionalResults) {
            return angular.extend(results, additionalResults);
        });
    })

    // ...
});
```

**Comment by AkA84** [comment]
Approved

**Pull request merged**